src/compas/robots/resources/basic.py

```diff
@@ -70,18 +92,49 @@ def can_load_mesh(self, url):
         is_local_file = len(scheme) in (0, 1) or scheme == 'file'
 
         if is_local_file:
-            return True
+            if os.path.isfile(url):
+                return True
 
         # Only OBJ loader supports remote files atm
         is_obj = _get_file_format(url) == 'obj'
         return scheme in ('http', 'https') and is_obj
 
     def load_mesh(self, url):
+        """Loads a mesh from local storage.
+
+        Parameters
+        ----------
+        url : str
+            Mesh location
+
+        Returns
+        -------
+        :class:`Mesh`
+            Instance of a mesh.
+        """
         if url.startswith('file:///'):
             url = url[8:]
 
         return _mesh_import(url, url)
 
+    def _get_local_path(self, url):
+        """Concatenates filepath directory to URL only if defined in the keyword arguments.
+
+        Parameters
+        ----------
+        url : str
+            Mesh location.
+
+        Returns
+        -------
+        url: str
+            Extended mesh location if filepath in kwargs. 
+            Else, ir returns None. 
+        """
+        filepath = self.attr.get('filepath') 
```

Comment:
How about `basepath` instead of `filepath` here?

Reply:
I was actually thinking about what convention to use regarding the `filepath` name. `basepath` sounds good to me, and I will actually use the same variable name to the dictionary entry as well as with the local variable (e.g. `basepath = self.attr.get('basepath')`).